src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,50 @@
 
 class Animal {
   // write your code here
+  // static alive = [];
+
+  constructor() {
+    this.health = 100;
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
   // write your code here
+  constructor() {
+    super();
+    this.hidden = false;
+
+    Animal.alive.push(this);
+  };
+
+  hide() {
+    this.hidden = true;
+  };
 }
 
 class Carnivore extends Animal {
   // write your code here
+  constructor() {
+    super();
+
+    Animal.alive.push(this);
+  };
+
+  bite(name) {
+    if (name.hidden === false
+      && name instanceof Herbivore
+      && name.health > 0) {
+      name.health -= 50;
+    }
```

Comment:
let's simplify our condition: you don't need verifying `health`, the following code will take care of it 
```suggestion
    if (!victim.hidden && victim instanceof Herbivore) {
      name.health -= 50;
    }
```